etl/etl.py

```diff
@@ -3,51 +3,105 @@
 import pathlib
 from pprint import pprint
 from contextlib import closing
+from typing import List, Optional
 
 from gremlin_python.process.anonymous_traversal import traversal
 from gremlin_python.driver.driver_remote_connection import DriverRemoteConnection
 from gremlin_python.process.graph_traversal import __
 
-def extract_from_nix():    
+
+def extract_from_nix():
     nixpkgs_graph_nix_file_path = (
         pathlib.Path(__file__).parent.joinpath("nixpkgs-graph.nix").resolve()
     )
-    print("Downloading data from nix as a dataframe...")    
-    result = subprocess.run('NIXPKGS_FLAKE_REF="github:nixos/nixpkgs/master" nix eval --json --file '+ str(nixpkgs_graph_nix_file_path) + ' --extra-experimental-features "nix-command flakes"', shell=True, stdout=subprocess.PIPE)
-    dataframe = pd.read_json(result.stdout.decode(), orient="records")    
+    print("Downloading data from nix as a dataframe...")
+    result = subprocess.run(
+        'NIXPKGS_FLAKE_REF="github:nixos/nixpkgs/master" nix eval --json --file '
+        + str(nixpkgs_graph_nix_file_path)
+        + ' --extra-experimental-features "nix-command flakes"',
+        shell=True,
+        stdout=subprocess.PIPE,
+    )
+    dataframe = pd.read_json(result.stdout.decode(), orient="records")
     return dataframe
 
-def data_process(dataframe):
-    dataframe["path"] = dataframe.groupby(["outputPath"])["path"].transform(lambda x : ", ".join(x))
+
+def data_process(dataframe: pd.DataFrame):
```

Comment:
Minor nitpick, but I would favor swapping the order of the function name to `process_data`. Generally speaking, I think using names which are actions (`do_this`, `get_that`) are favored for method / function names while nouns like this are more commonly used for objects, classes, etc.